docs/addons/nats/tls/index.md

```diff
@@ -0,0 +1,783 @@
+--set nats.tls.key="tls.key" \
+--set nats.tls.verify=true \
+--set cluster.enabled=true \
+--set cluster.replicas=3 
+
+```
+
+This chart will create the necessary StatefulSet, Service, PVCs etc. for the NATS cluster. You can easily view all the resources created by chart using [ketall](https://github.com/corneliusweig/ketall) `kubectl` plugin as below,
+
+```bash
+❯ kubectl get-all -n demo -l app.kubernetes.io/instance=sample-nats-tls
+NAME                                                            NAMESPACE  AGE
+configmap/sample-nats-tls-config                                demo       9m40s  
+endpoints/sample-nats-tls                                       demo       9m40s  
+persistentvolumeclaim/sample-nats-tls-js-pvc-sample-nats-tls-0  demo       9m40s  
+persistentvolumeclaim/sample-nats-tls-js-pvc-sample-nats-tls-1  demo       9m17s  
+persistentvolumeclaim/sample-nats-tls-js-pvc-sample-nats-tls-2  demo       8m54s  
+pod/sample-nats-tls-0                                           demo       9m40s  
+pod/sample-nats-tls-1                                           demo       9m17s  
+pod/sample-nats-tls-2                                           demo       8m54s  
+service/sample-nats-tls                                         demo       9m40s  
+controllerrevision.apps/sample-nats-tls-76dfb9c75               demo       9m40s  
+statefulset.apps/sample-nats-tls                                demo       9m40s  
+endpointslice.discovery.k8s.io/sample-nats-tls-6lxps            demo       9m40s 
+
+```
+
+Now, wait for the NATS server pods `sample-nats-tls-0`, `sample-nats-tls-1`, `sample-nats-tls-2` to go into `Running` state,
+
+```bash
+❯ kubectl get pod -n demo -l app.kubernetes.io/instance=sample-nats
+NAME                READY   STATUS    RESTARTS   AGE
+sample-nats-tls-0   3/3     Running   0          11m
+sample-nats-tls-1   3/3     Running   0          11m
+sample-nats-tls-2   3/3     Running   0          11m
+```
+
+Once the pods are in `Running` state, verify that the NATS server is ready to accept the connections.
+
+```bash
+❯ kubectl logs -n demo sample-nats-tls-0 -c nats
+[7] 2021/09/06 08:33:53.111508 [INF] Starting nats-server
+[7] 2021/09/06 08:33:53.111560 [INF] Version:  2.4.0
+...
+[7] 2021/09/06 08:33:53.116004 [INF] Server is ready
+```
+
+From the above log, we can see the NATS server is ready to accept connections.
+
+### Insert Sample Data
+The above Helm chart also deploy a pod with nats-box image which can be used to interact with the NATS server. Let's verify the nats-box pod has been created.
+
+```bash
+❯ kubectl get pod -n demo -l app=sample-nats-tls-box
+NAME                                   READY   STATUS    RESTARTS   AGE
+sample-nats-tls-box-67fb4fb4f9-gtt9z   1/1     Running   0          13m
+```
+
+Now, we are going to exec into the nats-box pod and create some sample data. The nats-client-tls certificate has created a secret with access credentials. Let's find out the credentials from the secret.
```

Comment:
Now, we are going to exec into the nats-box pod and create some sample data, We are going to use the client certificates created in `nats-client-tls` Secret to connect with the NATS server. So, let's create the certificates files inside the nats-box pod.

At first, let's get the certificates from the `nats-client-tls` Secret,